backend/src/sync.js

```diff
@@ -52,29 +52,44 @@ async function saveBlocks(blocksInfo) {
               return Promise.all([
                 models.Transaction.bulkCreate(
                   blockInfo.transactions.map(tx => {
-                    const actions = tx.actions.map(action => {
-                      if (typeof action === "string") {
-                        return { [action]: {} };
-                      }
-                      if (action.DeployContract !== undefined) {
-                        delete action.DeployContract.code;
-                      } else if (action.FunctionCall !== undefined) {
-                        delete action.FunctionCall.args;
-                      }
-                      return action;
-                    });
                     return {
                       hash: tx.hash,
                       nonce: tx.nonce,
                       blockHash: blockInfo.header.hash,
                       signerId: tx.signer_id,
                       signerPublicKey: tx.signer_public_key || tx.public_key,
                       signature: tx.signature,
-                      receiverId: tx.receiver_id,
-                      actions
+                      receiverId: tx.receiver_id
                     };
                   })
                 ),
+                blockInfo.transactions.map(tx => {
```

Comment:
JS fails you here. `Promise.all()` accepts a list of promises, and so we have a list of:

* one promise created with `models.Transaction.bulkCreate`
* a list of promises created for each `transaction` in `blockInfo.transactions` with `models.Action.bulkCreate`
* one promise created with `models.Account.bulkCreate`

`Promise.all` will consider the list of `models.Action.bulkCreate` promises as an already resolved value, and will not await for it.

Concatenate the list of `models.Transaction.bulkCreate` and `models.Account.bulkCreate` promises with the list of promises created with the `blockInfo.transactions.map`.

Example:

```js
async function qq1() {
  const x = Promise.all(
    [
      [new Promise((resolve) => setTimeout(() => { console.log('QQ1: resolved'); resolve() }, 1000))]
    ]
  );
  await x;
  console.log('QQ1: all the promises are resolved');
};

async function qq2() {
  const x = Promise.all(
    [
      new Promise((resolve) => setTimeout(() => { console.log('QQ2: resolved'); resolve() }, 1000))
    ]
  );
  await x;
  console.log('QQ2: all the promises are resolved');
};
```

When you run `qq1`, you will get:

```
QQ1: all the promises are resolved
QQ1: resolved
```

That is not what you want.

When you run `qq2`, you will get:

```
QQ2: resolved
QQ2: all the promises are resolved
```

Reply:
@icerove Please, familiarize yourself with Promises and async/await in JS. Here is what you essentialy have here now:

```
Promise.all(
  [
     [undefined, undefined, ...],
     Promise-to-bulk-create-Accounts
  ]
)
```

Well, we should also refactor this code (break it apart).